apps/dashboard/src/components/workflow-editor/workflow-checklist.tsx

```diff
@@ -59,26 +64,35 @@ export function WorkflowChecklist({ steps, workflow }: WorkflowChecklistProps) {
             unsafeMetadata: {
               ...user.unsafeMetadata,
               workflowChecklistCompleted: true,
+              workflowChecklistClosed: true,
             },
           });
         }
-      } else {
+      } else if (!user?.unsafeMetadata?.workflowChecklistClosed) {
         setIsOpen(true);
       }
     }
   }, [steps, checklistItems, currentEnvironment, workflow, integrations, environments, user, telemetry]);
 
   const handleOpenChange = (open: boolean) => {
-    if (open === false) return;
     setIsOpen(open);
 
-    telemetry(TelemetryEvent.WORKFLOW_CHECKLIST_OPENED, {
-      workflowId: workflow?.workflowId,
-    });
+    if (open) {
+      telemetry(TelemetryEvent.WORKFLOW_CHECKLIST_OPENED, {
+        workflowId: workflow?.workflowId,
+      });
+    } else {
+      user?.update({
+        unsafeMetadata: {
+          ...user.unsafeMetadata,
+          workflowChecklistClosed: true,
+        },
+      });
+    }
   };
 
   return (
-    <Popover open={isOpen} onOpenChange={handleOpenChange}>
+    <Popover open={isOpen} onOpenChange={handleOpenChange} modal={false}>
```

Comment:
When we pass `modal={false}`, it's easier to control the open state, for example, I've improved it with having a toggle behaviour on the pill and ignoring the outside clicks.